components/login/button.tsx

```diff
@@ -1,15 +1,38 @@
 "use client";
 
-import { signIn } from "next-auth/react";
-import Image from "next/image";
-import Link from "next/link";
+import { signIn, useSession } from "next-auth/react";
+import Login from "@/public/login.svg";
+import { MenuContent } from "@/types/ui-content";
+import { ProfileModal } from "../layout/navbar/profile-menu/profile-modal";
+import { handleLogin } from "../layout/navbar/profile-menu/handle-login";
+import { useEffect } from "react";
+
+type LoginProps = {
+  profileMenu: MenuContent[];
+  domain: string;
+};
+
+export const LoginButton = ({ domain, profileMenu }: LoginProps) => {
+  const { data: session, status } = useSession();
+  useEffect(() => {
+    if (status === "authenticated") {
+      handleLogin(domain, session);
+    }
+  }, [status, session, domain]);
+
+  const initiateLogin = async () => {
+    await signIn("google");
+  };
 
-export const LoginButton = () => {
   return (
-    <div className="relative z-50 h-12 w-12">
-      <button onClick={() => signIn("google")}>
-        <Image alt="login" src="/login.svg" fill />
-      </button>
+    <div className="h-10 w-10">
+      {session ? (
+        <ProfileModal menu={profileMenu} domain={domain} />
+      ) : (
+        <button onClick={initiateLogin}>
```

Comment:
button as no call is this the div necessary then in our case?
I feel like you could do something like that
```
<>
  {session ? (
         <ProfileModal menu={profileMenu} domain={domain} />
       ) : (
         <button onClick={initiateLogin} className="h-10 w-10">
          <Login className="h-10 w-10 !fill-none" />
         </button>
       )}
<>
```

Reply:
you/re right, i changed it